src/database/listRemote.ts

```diff
@@ -0,0 +1,82 @@
+import * as request from "request";
+import { Response } from "request";
+import * as responseToError from "../responseToError";
+import * as utils from "../utils";
+import * as FirebaseError from "../error";
+import * as logger from "../logger";
+import * as api from "../api";
+
+export interface ListRemote {
+  /**
+   * Call the shallow get API with limitToFirst=numSubPath.
+   * @param path the path to list
+   * @param numSubPath the number of subPaths to fetch.
+   * @return the list of sub pathes found.
+   */
+  listPath(
+    path: string,
+    numSubPath: number,
+    startAfter?: string,
+    timeout?: number
+  ): Promise<string[]>;
+}
+
+export class RTDBListRemote implements ListRemote {
+  private instance: string;
+
+  constructor(instance: string) {
+    this.instance = instance;
+  }
+
+  async listPath(
+    path: string,
+    numSubPath: number,
+    startAfter?: string,
+    timeout?: number
+  ): Promise<string[]> {
+    const startAfterSuffix = startAfter ? "&startAfter=" + startAfter : "";
+    const timeoutSuffix = timeout ? "&timeout=" + timeout + "ms" : "";
+    const url =
+      utils.addSubdomain(api.realtimeOrigin, this.instance) +
+      path +
+      `.json?shallow=true&limitToFirst=${numSubPath}` +
+      startAfterSuffix +
+      timeoutSuffix;
```

Comment:
I think this will be easier to read if we use template strings for the whole thing

Reply:
agree